skyllh/analyses/i3/publicdata_ps/signal_generator.py

```diff
@@ -0,0 +1,563 @@
+        dataset_list : list of Dataset instances
+            The list of Dataset instances for which signal events should get
+            generated for.
+        data_list : list of DatasetData instances
+            The list of DatasetData instances holding the actual data of each
+            dataset. The order must match the order of ``dataset_list``.
+        llhratio : LLHRatio
+            The likelihood ratio object contains the datasets signal weights
+            needed for distributing the event generation among the different
+            datsets.
+        energy_cut_splines : dict
+        cut_energy_min : dict
+        """
+        self.src_hypo_group_manager = src_hypo_group_manager
+        self.dataset_list = dataset_list
+        self.data_list = data_list
+        self.llhratio = llhratio
+        self.effA = [None] * len(self._dataset_list)
+        self.sm = [None] * len(self._dataset_list)
+        self.splines = energy_cut_splines
+        self.cut_sindec = cut_sindec
+
+    @property
+    def src_hypo_group_manager(self):
+        """The SourceHypoGroupManager instance defining the source groups with
+        their spectra.
+        """
+        return self._src_hypo_group_manager
+
+    @src_hypo_group_manager.setter
+    def src_hypo_group_manager(self, manager):
+        if(not isinstance(manager, SourceHypoGroupManager)):
+            raise TypeError('The src_hypo_group_manager property must be an '
+                            'instance of SourceHypoGroupManager!')
+        self._src_hypo_group_manager = manager
+
+    @property
+    def dataset_list(self):
+        """The list of Dataset instances for which signal events should get
+        generated for.
+        """
+        return self._dataset_list
+
+    @dataset_list.setter
+    def dataset_list(self, datasets):
+        if(not issequenceof(datasets, Dataset)):
+            raise TypeError('The dataset_list property must be a sequence of '
+                            'Dataset instances!')
+        self._dataset_list = list(datasets)
+
+    @property
+    def llhratio(self):
+        """The log-likelihood ratio function for the analysis.
+        """
+        return self._llhratio
+
+    @llhratio.setter
+    def llhratio(self, llhratio):
+        if llhratio is not None:
+            if(not isinstance(llhratio, LLHRatio)):
+                raise TypeError('The llratio property must be an instance of '
+                                'LLHRatio!')
+        self._llhratio = llhratio
+
+    @property
+    def sig_gen_list(self):
+        """The list of PublicDataDatasetSignalGenerator instances for each dataset
+        """
+        return self._sig_gen_list
+
+    @sig_gen_list.setter
+    def sig_gen_list(self, sig_gen_list):
+        if(not issequenceof(sig_gen_list, PDDatasetSignalGenerator)):
+            raise TypeError('The sig_gen_list property must be a sequence of '
+                            'PublicDataDatasetSignalGenerator instances!')
```

Comment:
It does not really make sense to implement saving sig_gen_list now. In case of a slow signal injection, may be worth revisiting after the skyllh v2 update.

Deleted in https://github.com/icecube/skyllh/commit/a96eba1c33a790b862fa9d5ad414f3853c31a25d